keps/sig-instrumentation/2305-metrics-cardinality-enforcement/README.md

```diff
@@ -236,19 +264,145 @@ This would then be interpreted by our machinery as this:
 
 ```
 
-## Open-Question
-
+## Design Details
+### Test Plan
+Unit test to verify that the metric label will be set to "unexpected" if the metric encounters label values outside our explicit allowlist of values.
+### Graduation Criteria
+The allowlist of metrics can be configured via the exposed flag and the unit test is passed.
+### Upgrade / Downgrade strategy
+N/A
+### Version Skew Strategy
+N/A
+
+## Production Readiness Review Questionnaire
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: The metric cardinality will be enforced by default. The allowlist of label values for a metric is dynamically configured with flag.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? No
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled). No
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+  No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g., if this is a runtime
+  feature, can it break the existing applications?).
+  No. The metric cardinality enforcement will be enabled by default.
+  
+* **What happens if we reenable the feature if it was previously rolled back?**
+  There is no functionality concern about enable/disable of the feature.
+  
+* **Are there any tests for feature enablement/disablement?**
+  No.  
```

Comment:
This will need tests.

Reply:
do you have any example of what tests are needed here?

Reply:
You would need to test for the behaviour when the feature is on vs. off, as well as what happens when you disable a feature flag in a live component typically. Not all of this would be automated.

Reply:
@ehashman said it well

Reply:
ok, changed.